### ISSUE TITLE
Expo AuthSession sdk33 dismissBrowser missing on android

### ISSUE THREAD

**Issue by andnilsson** [expo/expo#4559]
### Environment

Expo CLI 2.19.5 environment info:
    System:
      OS: Windows 10
    Binaries:
      Yarn: 1.5.1 - C:\Users\AndreasNilsson\AppData\Roaming\npm\yarn.CMD
      npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.1.0.0 AI-173.4819257

Targets android, ios

### Steps to Reproduce

Run the code. I´m using Microsoft B2CAD as ouath2 provider

### Expected Behavior

The auth session flow should return a promise with autorization data

### Actual Behavior

The flow just stops when redirecting back to the app

This code works for me on SDK 32.
After migrating to SDK 33, on the emulator i just get the screen with the "Return to the app->" button, and nothing happends when i press it. I don´t get any console output either

On a physical device (still in the expo client) I  get this warning:
![image](https://user-images.githubusercontent.com/25838327/59328078-c4e42500-8ceb-11e9-8687-de8fbbe4fbaa.png)

It works perfectly fine on ios emulator

### Reproducible Demo

```
import React from 'react';
import { Text, Button, View } from 'react-native';
import { AuthSession, Linking } from 'expo';
import config from './config'


export default class App extends React.Component {
	state = {
		result: null
	};

	authorize = async () => {
		console.log('starting auth flow')
		let redirectUrl = encodeURIComponent(AuthSession.getRedirectUrl());
		let noise = "asdf"
		let authUrl = `https://login.microsoftonline.com/${config.tenant}/oauth2/authorize?client_id=${config.client_id}&response_type=code&redirect_uri=${redirectUrl}&response_mode=query&resource=${config.api_resource_uri}&state=${noise}`
		let result = await AuthSession.startAsync({ authUrl, returnUrl: Linking.makeUrl() });

		console.log(result)
		this.setState({ result });
	}

	render() {
		return (
			<View>
				<Button title="authorize" onPress={this.authorize} />
				{this.state.result ? (
					<Text>{JSON.stringify(this.state.result)}</Text>
				) : null}
			</View>
		)

	}
}
```

**Comment by cruzach** [comment]
If you read through [this PR](https://github.com/expo/expo/pull/4473), this is something we need to revisit, as we only have a temporary fix out right now.

**Comment by brentvatne** [comment]
hey there! a couple of things:

- `WebBrowser.dismissBrowser()` prior to SDK33 was a noop - it wouldn't actually do anything but it wouldn't complain. Now it throws an error to let you know that it will not do anything.
- When the web browser window doesn't close after redirect you're encountering this issue: https://github.com/expo/expo/issues/4524 - don't worry, it doesn't happen in standalone Android apps so it won't impact your project in production. It only happens in the Expo client.

We'll keep tracking this in https://github.com/expo/expo/issues/4524

**Comment by jpagand** [comment]
Did you manage to make it work in standalone Android apps?
`AuthSession.startAsync` is rejected on SDK33 even with this fix #4473
